File: src/main/java/xyz/ivyxjc/pubg4j/entity/PubgPlayer.java
package xyz.ivyxjc.pubg4j.entity;

import java.time.LocalDateTime;
import java.util.List;
import lombok.AccessLevel;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import xyz.ivyxjc.pubg4j.types.PlatformRegion;

/**
 * @author Ivyxjc
 * @since 4/22/2018
 */

@Data
public class PubgPlayer {
    /**
     * Player ID
     */
    private String playerId;
    private LocalDateTime createdAt;
    private String name;

    /**
     * Platform-region shard
     */
    private PlatformRegion shardId;

    /**
     * Identifies the studio and game
     */
    private String titleId;

    private LocalDateTime updatedAt;

    private List<PubgMatch> matches;

    @Getter(AccessLevel.NONE)
    @Setter(AccessLevel.NONE)
    private String patchVersion;

    @Getter(AccessLevel.NONE)
    @Setter(AccessLevel.NONE)
    private String stats;
}

File: src/main/kotlin/xyz/ivyxjc/pubg4j/entity/PubgPlayer.kt
package xyz.ivyxjc.pubg4j.entity

import xyz.ivyxjc.pubg4j.types.PlatformRegion
import java.time.LocalDateTime
import java.util.*

/**
 * @author Ivyxjc
 * @since 4/22/2018
 */

class PubgPlayer {

    constructor()

    constructor(playerId: String) {
        this.playerId = playerId
    }

    lateinit var playerId: String

    var createdAt: LocalDateTime? = null

    var name: String? = null

    /**
     * Platform-region shard
     */
    var shardId: PlatformRegion? = null

    /**
     * Identifies the studio and game
     */
    var titleId: String? = null

    var updatedAt: LocalDateTime? = null

    val matches: MutableList<PubgMatch>

    private val patchVersion: String? = null

    private val stats: String? = null

    init {
        matches = ArrayList()
    }

    override fun toString(): String {
        return "PubgPlayer(playerId='$playerId', createdAt=$createdAt, name=$name, shardId=$shardId, titleId=$titleId, updatedAt=$updatedAt, matches=$matches, patchVersion=$patchVersion, stats=$stats)"
    }

}
